es-ds-components/components/es-video.vue

```diff
@@ -0,0 +1,73 @@
+<script setup lang="ts">
+import { useElementSize } from '@vueuse/core';
+
+interface IProps {
+    altText: string;
+    coverImageUrl: string;
+    embedUrl: string;
+}
+const props = defineProps<IProps>();
+const showVideo = ref(false);
+function getVideoIdFromUrl(url: string) {
+    const embedPattern = /https:\/\/www\.youtube\.com\/embed\/(?<videoId>[A-Za-z0-9-]+)/;
+    const parsedUrl = embedPattern.exec(url);
+    if (parsedUrl) {
+        return parsedUrl?.groups?.videoId;
+    }
+    return '';
+}
+const videoId = getVideoIdFromUrl(props.embedUrl);
+const embedUrlWithParams = `https://www.youtube.com/embed/${videoId}?rel=0&autoplay=1&cc_load_policy=1&cc_lang_pref=en`;
+
+const placeholderCardEl = useTemplateRef('placeholder-card');
+const cardDimensionElementSize: { width: number; height: number } = reactive(
+    useElementSize(placeholderCardEl, { width: 0, height: 0 }, { box: 'border-box' }),
+);
```

Comment:
At the moment, when the iframe is loaded on mobile, it stops retaining its aspect ratio if the window is resized, whereas the old EsVideo did retain its aspect ratio. I think we should try to keep this behavior.

The `<b-embed>` component uses a few simple CSS classes to handle keeping the 16:9 ratio we want; I think we should copy those over and use those as a simpler solution than live-listening to the size of the card and updating a bunch of JS variables.

These CSS classes already appear in our es-ds-styles in the [embed utilities SASS file](https://github.com/EnergySage/es-ds/blob/esds-3.0-vue3-primevue/es-ds-styles/scss/utilities/_embed.scss), so they're already there for us to use. We just need to do [what <b-embed> does](https://github.com/bootstrap-vue/bootstrap-vue/blob/dev/src/components/embed/embed.js#L39) and put the associated classes on the parent element of the iframe.

Reply:
Ahh yeah CSS solution is much better 👍